fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/SimplifyArithmeticComparisonRule.java

```diff
@@ -40,68 +58,90 @@
-                    Expression tmp = right;
-                    right = left;
-                    left = tmp;
-                }
+            try {
+                List<Expression> children = tryRearrangeChildren(newComparison.left(), newComparison.right());
+                newComparison = (ComparisonPredicate) newComparison.withChildren(children);
+            } catch (Exception e) {
+                return comparison;
             }
         }
-        if (left != predicate.left() || right != predicate.right()) {
-            predicate = (ComparisonPredicate) predicate.withChildren(left, right);
-            return TypeCoercionUtils.processComparisonPredicate(predicate);
-        } else {
-            return predicate;
-        }
+        return TypeCoercionUtils.processComparisonPredicate(newComparison);
     }
 
-    @Override
-    public Expression visitGreaterThan(GreaterThan greaterThan, ExpressionRewriteContext context) {
-        return process(greaterThan);
+    private boolean couldRearrange(ComparisonPredicate cmp) {
+        return rearrangementMap.containsKey(cmp.left().getClass())
+                && !cmp.left().isConstant()
+                && cmp.left().children().stream().anyMatch(Expression::isConstant);
     }
 
-    @Override
-    public Expression visitGreaterThanEqual(GreaterThanEqual greaterThanEqual, ExpressionRewriteContext context) {
-        return process(greaterThanEqual);
-    }
+    private List<Expression> tryRearrangeChildren(Expression left, Expression right) throws Exception {
+        if (!left.child(1).isLiteral()) {
+            throw new RuntimeException(String.format("Expected literal when arranging children for Expr %s", left));
+        }
+        Literal leftLiteral = (Literal) left.child(1);
+        Expression leftExpr = left.child(0);
 
-    @Override
-    public Expression visitEqualTo(EqualTo equalTo, ExpressionRewriteContext context) {
-        return process(equalTo);
-    }
+        Class<? extends Expression> oppositeOperator = rearrangementMap.get(left.getClass());
+        Expression newChild = oppositeOperator.getConstructor(Expression.class, Expression.class)
+                .newInstance(right, leftLiteral);
 
-    @Override
-    public Expression visitLessThan(LessThan lessThan, ExpressionRewriteContext context) {
-        return process(lessThan);
+        if (left instanceof Divide && leftLiteral.compareTo(new IntegerLiteral(0)) < 0) {
+            // Multiplying by a negative number will change the operator.
+            return Arrays.asList(newChild, leftExpr);
+        }
+        return Arrays.asList(leftExpr, newChild);
     }
 
-    @Override
-    public Expression visitLessThanEqual(LessThanEqual lessThanEqual, ExpressionRewriteContext context) {
-        return process(lessThanEqual);
+    // Ensure that the second child must be Literal, such as
+    private @Nullable ComparisonPredicate normalize(ComparisonPredicate comparison) {
```

Comment:
It would be better to make it public, because we could use it to generate more effecient Runtime filter